Add documentation guidelines when adding new APIs (#1335)
dotnet/runtime: docs/coding-guidelines/adding-api-guidelines.md

  - Add new test code following [conventions](project-guidelines.md#code-file-naming-conventions) for new files to that are specific to the new target framework.
  - To run just the new test configuration run `dotnet msbuild <Library>.csproj /t:RebuildAndTest /p:TargetGroup=<TargetGroup>`

## Documentation

> New public APIs must be documented with triple slash comments on top of them. Visual Studio automatically generates the structure for you when you type `\\\`.

Reviewer: I thought `///` is what Visual Studio recognized for automatically generating the structure.

Reviewer 2: ```suggestion
New public APIs must be documented with triple-slash comments on top of them. Visual Studio automatically generates the structure for you when you type `///`.
```

If your new API or the APIs it calls throw any exceptions, those need to be manually documented by adding the `<exception></exception>` elements.

> After your change is merged, we will eventually port them to the dotnet-api-docs repo, where we will review them for language and proper style (See the [Docs contributing guide](https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md) for more information).

Reviewer 2: It would probably be best to link to https://github.com/dotnet/dotnet-api-docs/wiki instead.

```suggestion
After your change is merged, we will eventually port them to the dotnet-api-docs repo, where we will review them for language and proper style (For more information, see the [API writing guidelines](https://github.com/dotnet/dotnet-api-docs/wiki)).
```

> Once the dotnet-api-docs change is merged, your comments will start showing up in MS Docs, and later in IntelliSense.

Reviewer 3: ```suggestion
Once the dotnet-api-docs change is merged, your comments will start showing up in the official API documentation at http://docs.microsoft.com/, and later it will appear in IntelliSense in Visual Studio and Visual Studio Code.
```

> 

Reviewer 3: ```suggestion
Once the documentation is official, any subsequent updates to it must be made directly in https://github.com/dotnet/dotnet-api-docs/. It's fine to make updates to the triple slash comments later, they just won't automatically flow into the official docs.
```

> 

Reviewer 3: Once the documentation is official, any subsequent updates to it must be made directly in https://github.com/dotnet/dotnet-api-docs/. You can update the triple slash comments, but those later updates won't be ported there.

> Once the dotnet-api-docs change is merged, your comments will start showing up in the official API documentation at http://docs.microsoft.com/, and later it will appear in IntelliSense in Visual Studio and Visual Studio Code.

Reviewer 2: ```suggestion
Once the dotnet-api-docs change is merged, your comments will start showing up in the official API documentation at http://docs.microsoft.com/, and later they'll appear in IntelliSense in Visual Studio and Visual Studio Code.
```

Reviewer 2: And VS for Mac too, no?

Author: I suppose we will see it there as well. Saying simply "Visual Studio" sounds like it covers both Windows and Mac.
